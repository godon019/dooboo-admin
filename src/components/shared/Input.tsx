import styled, { CSSProp } from 'styled-components';

import React from 'react';

interface InputProps {
  name: string;
  type: string;
  unit?: string;
  value?: any;
  readOnly?: boolean;
  customCss?: CSSProp;
}

const Container = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  border: 1px solid #d9d9da;
`;

const CustomInput = styled.input<{ addCSS: CSSProp }>`
  width: 100%;
  /* To fix UI abmormality, commented this */
  /* height: 60px; */
  padding: 1rem 0.8rem;
  font-size: 1rem;
  color: ${(props) => (props.readOnly ? 'grey' : '#000000')};
  ${(props) => props.addCSS};
  border: none;
  &:focus {
    outline: none;
  }
  &::placeholder {
    color: rgb(215, 216, 217);
    font-size: 14px;
  }
`;

const UnitSpan = styled.span`
  text-align: center;
  line-height: 60px;
  height: 60px;
  padding: 0 15px;
`;

const Input: React.FC<InputProps & React.HTMLAttributes<HTMLInputElement>> = ({
  name,
  type,
  readOnly,
  value,
  customCss,
  unit,
  ...otherProps
}) => {
  return (
    <Container>
      <CustomInput
        name={name}
        type={type}
        readOnly={readOnly}
        value={value}
        addCSS={customCss}
        {...otherProps}
      />
      {unit ? <UnitSpan>{unit}</UnitSpan> : null}
    </Container>
  );
};

export default Input;
