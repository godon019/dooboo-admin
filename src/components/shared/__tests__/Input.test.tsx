import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../Input';
import { render } from '@testing-library/react';

let props: any;

props = {
  name: 'name',
  type: 'text',
};

let rendered;
let utils;

beforeAll(() => {
  rendered = renderer.create(<Input {...props}/>).toJSON();
  utils = render(<Input {...props}/>);
});

describe('[Input] render', () => {
  it('renders without crashing', () => {
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});
