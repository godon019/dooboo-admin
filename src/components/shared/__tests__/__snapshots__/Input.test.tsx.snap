// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Input] render renders without crashing 1`] = `
<div
  className="sc-bdVaJa elFmeq"
>
  <input
    className="sc-bwzfXH faoLVF"
    name="name"
    type="text"
  />
</div>
`;
